test/config-luatest/credentials_applier_test.lua

```diff
@@ -783,3 +783,255 @@ g.test_sync_ro_rw = function(g)
+                                {execute = true})
+            t.assert_equals(perm_2['function']['myfunc2'], nil)
+            t.assert_equals(perm_1['sequence']['myseq3'], nil)
+            t.assert_equals(perm_2['sequence']['myseq3'],
+                                {read = true, write = true})
+
+            t.assert_equals(perm_1['space']['myspace2'], nil)
+            t.assert_equals(perm_2['space']['myspace3'], nil)
+            t.assert_equals(perm_1['function']['myfunc1'], nil)
+            t.assert_equals(perm_2['function']['myfunc3'], nil)
+            t.assert_equals(perm_1['sequence']['myseq1'], nil)
+            t.assert_equals(perm_2['sequence']['myseq1'], nil)
+        end,
+        options_2 = iconfig,
+        verify_2 = function(iconfig)
+            box.schema.space.create('myspace2')
+            box.schema.space.create('myspace3')
+            box.schema.func.create('myfunc1')
+            box.schema.func.create('myfunc3')
+            box.schema.sequence.create('myseq1')
+            box.schema.sequence.create('myseq2')
+
+            local internal =
+                    require('internal.config.applier.credentials')._internal
+
+            local perm_1 = box.schema.user.info('myuser1')
+            perm_1 = internal.privileges_from_box(perm_1)
+
+            local perm_2 = box.schema.user.info('myuser2')
+            perm_2 = internal.privileges_from_box(perm_2)
+
+            local exp_perm_1 = iconfig.credentials.users.myuser1
+            exp_perm_1 = internal.privileges_from_config(exp_perm_1)
+            exp_perm_1 = internal.privileges_add_defaults('myuser1', 'user',
+                                                          exp_perm_1)
+
+            local exp_perm_2 = iconfig.credentials.users.myuser2
+            exp_perm_2 = internal.privileges_from_config(exp_perm_2)
+            exp_perm_2 = internal.privileges_add_defaults('myuser2', 'user',
+                                                          exp_perm_2)
+
+            t.assert_equals(perm_1, exp_perm_1)
+            t.assert_equals(perm_2, exp_perm_2)
+        end,
+        verify_args_2 = {iconfig},
+    })
+end
+
+g.test_postpone_grants_till_rename = function(g)
+    local iconfig = {
+        credentials = {
+            users = {
+                guest = {
+                    roles = { 'super' }
+                },
+                myuser1 = {
+                    privileges = {
+                        {
+                            permissions = {
+                                'read',
+                                'write',
+                            },
+                            spaces = {
+                                'myspace1',
+                                'myspace2',
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    }
+
+    -- Create myspace1, check, rename to myspace2, check, create myspace1,
+    -- check.
```

Comment:
TBH, I doesn't spot the key point of the test. I'll try to ask several questions around in order to understand it better.

There is a configuration reloading before the second creating of myspace1. Is it important for the test case? If so, it should be mentioned in the quoted list of actions. If not, `success_case` can be used. Is it correct?

What is the key difference of this test case from the next test case (which uses `success_case`)?

Maybe it worth to mention possible negative scenario: what exactly shouldn't go in a wrong way. For example, write not just 'check', but 'check that <...>'.

Reply:
> There is a configuration reloading before the second creating of myspace1. Is it important for the test case? If so, it should be mentioned in the quoted list of actions. If not, success_case can be used. Is it correct?

Yes, it is important. This is the scenario I told you about previously. We create an object1, it gets its privs, we rename it to object2 and it tries to apply colliding set of privs or just twice, depending on implementation. In my implementation it checks whether the list of actions is cleared on reload and if the `if_not_exists` inside trigger action is `true`.

Reply:
> What is the key difference of this test case from the next test case (which uses success_case)?

The next test case is straightforward rename, not a corner case. I felt that it is not well covered by the previous one.

Reply:
Added `reload` to the comment.

Reply:
So, in brief, this scenario attempts to trigger 'already exists' error on a privilege granting on `:reload()`. OK, got it.

IMHO, it is hard to deduce from the code itself, so worth to clarify in a comment.

I'm not insisting about test cases, as usual, but you can use my feeling to guess how another developer will read it.

Reply:
Updated the comments with info about what they test instead of how they test it.